python/pyspark/sql/tests.py

```diff
@@ -3383,6 +3400,43 @@ def test_vectorized_udf_varargs(self):
         res = df.select(f(col('id')))
         self.assertEquals(df.collect(), res.collect())
 
+    def test_vectorized_udf_timestamps(self):
+        from pyspark.sql.functions import pandas_udf, col
+        from datetime import date, datetime
+        schema = StructType([
+            StructField("idx", LongType(), True),
+            StructField("date", DateType(), True),
+            StructField("timestamp", TimestampType(), True)])
+        # TODO Fails with time before epoch: (0, date(1969, 1, 1), datetime(1969, 1, 1, 1, 1, 1))
```

Comment:
Somewhere in the tz_convert process, a timestamp before the epoch causes this to fail - not sure where yet